### PR TITLE
trigger resize on update if needed

### LINKED ISSUE #5641: Resizing canvas behind sprite texture regression in v5
### Expected Behavior
Resizing a PIXI.Sprite to match a PIXI.Texture matching an underlying HTML5 Canvas.

v4.5.3 expected visual output

### Current Behavior
In v4, whether intentionally or not, it visually works as expected.  In v5 the functionality has regressed and the baseTexture size is not updated causing distortion in the image.

v5.0.0 visual regression

### Steps to Reproduce
These are the steps currently in use for resizing; width is protected against being zero otherwise Chrome raises many errors.
```
let canvas = document.createElement("canvas");
let sprite = new PIXI.Sprite.from(canvas);
...
canvas.width = new_width > 0 ? new_width : 2;
sprite.texture.orig.width = new_width;
sprite.width = new_width;
sprite.texture.update();
```

The value of `new_width` may be fractional.

### Environment
- pixi.js 4.5.3, and 5.0.0.
- Chrome 76.
- Demo with 5.0.0: http://ahyoomee.miru.hk/zignage/not-so-tiny-ticker.html
- Demo with 4.5.3: http://ahyoomee.miru.hk/zignage-previous/not-so-tiny-ticker.html

### PR DESCRIPTION
**Pull request by ivanpopelyshev**
Fixes #5641 and #5609

I didnt test it yet.

~I'd like to add deprecation for `BaseTexture.source`.~

### CODE REVIEW

**Comment by ivanpopelyshev** [comment]
Haha, my wife computer has different credentials for git

**Comment by bigtimebuddy** [comment]
@ivanpopelyshev moving this to **master**

**Comment by ivanpopelyshev** [comment]
Now it works : https://www.pixiplayground.com/#/edit/nptPNFqJj~s3y3rv3oSIi

**Comment by ivanpopelyshev** [comment]
In v4 noFrame option never was true after constructor and texture auto resize worked only because `onBaseTextureUpdate` worked for all textures, which is wrong.

Now I preserve noFrame flag. We can tidy the code later.

File: packages/core/src/textures/Texture.js [hunk 1]
```diff
@@ -143,13 +143,12 @@ export default class Texture extends EventEmitter
         {
             if (this.noFrame)
             {
-                frame = new Rectangle(0, 0, baseTexture.width, baseTexture.height);
+                this.frame = new Rectangle(0, 0, baseTexture.width, baseTexture.height);
+                this.noFrame = true;
```
[the remark this answers is not in the document]
Comment (ivanpopelyshev) [reply packages/core/src/textures/Texture.js]: Frame setter sets it to false. Id like to refactor it, if you are ready to review. Im gonna add tests in a couple of hours anyway

[the remark this answers is not in the document]
Comment (ivanpopelyshev) [reply packages/core/src/textures/Texture.js]: Invariant: noFrame is true if there was no frame param in constructor, and frame setter wasnt called from outside.

**Pull request merged**